python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -9083,16 +9083,30 @@ def test_diff_dataframe_invalid_axis():
             "string_col": ["a", "b", "c", "d", "e"],
         },
         ["a", "b", "c", "d", "e"],
-        [np.nan, None, np.nan, None],
     ],
 )
-def test_diff_dataframe_non_numeric_dypes(data):
+def test_diff_unsupported_dtypes(data):
     gdf = cudf.DataFrame(data)
     with pytest.raises(
-        NotImplementedError,
-        match="DataFrame.diff only supports numeric dtypes",
+        TypeError,
+        match=r"unsupported operand type\(s\)",
     ):
-        gdf.diff(periods=2, axis=0)
+        gdf.diff()
+
+
+def test_diff_many_dtypes():
+    pdf = pd.DataFrame(
+        {
+            "dates": pd.date_range("2020-01-01", "2020-01-06", freq="D"),
+            "bools": [True, True, True, False, True, True],
+            "floats": [1.0, 2.0, 3.5, 4.0, 5.0, -1.7],
```

Comment:
Adding null or np.nan values in integer and float would be helpful to cover more tests scenarios

Reply:
```suggestion
            "floats": [1.0, 2.0, 3.5, np.nan, 5.0, -1.7],
```

Reply:
I'm not sure if we provide compatibility with Pandas' nullable integer type... but we can add a `np.nan` value to the float column.